Source/WebCore/layout/formattingContexts/inline/InlineLineBuilder.cpp

```diff
@@ -1078,6 +1118,40 @@ bool LineBuilder::tryPlacingFloatBox(const InlineItem& floatItem, LineBoxConstra
     auto& floatBox = floatItem.layoutBox();
     ASSERT(formattingState());
     auto& boxGeometry = formattingState()->boxGeometry(floatBox);
+    auto containingBlockMarginTrim = floatBox.parent().style().marginTrim();
+    auto floatingContext = FloatingContext { formattingContext(), *floatingState() };
+    auto trimMarginsForPositionedFloat = [&](LayoutPoint& floatPosition) {
+        auto marginStart = boxGeometry.marginStart();
+        auto marginEnd = boxGeometry.marginEnd();
+        if (floatingContext.isFloaingCandidateLogicallyLeftPositioned(floatBox)) {
```

Comment:
floatingContext.isFloaingCandidateLogicallyLeftPositioned is based on the FloatingState inline direction which is not necessarily the same as the IFC's inline direction (e.g.  float box is inside a LTR block container but it shares a FloatingState with intrusive floats coming from an RTL ancestor block container) and margin-trim should always use containing block context.